pkg/frontend/adminactions/resources_list.go

```diff
@@ -44,35 +60,48 @@ func (a *azureActions) ResourcesList(ctx context.Context) ([]byte, error) {
 			a.log.Warnf("API version not found for type %q", *res.Type)
 			continue
 		}
+
+		if hasWritten {
+			writer.Write(comaAsByteArray)
+		}
+		hasWritten = true
+
 		switch *res.Type {
 		case "Microsoft.Compute/virtualMachines":
 			vm, err := a.virtualMachines.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			writeObject(writer, arm.Resource{
 				Resource: vm,
 			})
 		default:
 			gr, err := a.resources.GetByID(ctx, *res.ID, apiVersion)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			writeObject(writer, arm.Resource{
 				Resource: gr,
 			})
 		}
 	}
 
-	return json.Marshal(armResources)
+	_, err = writer.Write([]byte("]"))
+
+	return err
+}
+
+func writeObject(writer io.Writer, resource arm.Resource) {
```

Comment:
We could probably return the error from `MarshallJSON` here just in case a single resource fails to marshall.

Question for my own knowledge - is the individual marshall of each resource required, or could we skip it and pass resources directly to `writer.Write`?

Reply:
then we would log it but not fail as it was before ? Don't see any problem with that. 

yes, the marshaling is required, writer.Write() only accepts `[]byte` , and also go doesn't make any assumption on how you want to convert it

Reply:
then we would log it but not fail as it was before ? Don't see any problem with that.

yes, the marshaling is required, writer.Write() only accepts []byte , and also go doesn't make any assumption on how you want to convert it